Tools/LyTestTools/ly_test_tools/environment/file_system.py

```diff
@@ -293,61 +293,75 @@ def delete(file_list, del_files, del_dirs):
     return True
 
 
-def create_backup(source, backup_dir):
+def create_backup(source, backup_dir, backup_name=None):
     """
     Creates a backup of a single source file by creating a copy of it with the same name + '.bak' in backup_dir
     e.g.: foo.txt is stored as backup_dir/foo.txt.bak
+    If backup_name is provided, it will create a copy of the source file named "backup_name + .bak" instead.
 
     :param source: Full path to file to backup
     :param backup_dir: Path to the directory to store backup.
+    :param backup_name: [Optional] Name of the backed up file to use instead or the source name.
     """
 
     if not backup_dir or not os.path.isdir(backup_dir):
         logger.error(f'Cannot create backup due to invalid backup directory {backup_dir}')
-        return
+        return False
 
     if not os.path.exists(source):
         logger.warning(f'Source file {source} does not exist, aborting backup creation.')
-        return
+        return False
 
-    source_filename = os.path.basename(source)
-    dest = os.path.join(backup_dir, f'{source_filename}.bak')
+    dest = None
+    if backup_name is None:
+        source_filename = os.path.basename(source)
+        dest = os.path.join(backup_dir, f'{source_filename}.bak')
+    else:
+        dest = os.path.join(backup_dir, f'{backup_name}.bak')
 
     logger.info(f'Saving backup of {source} in {dest}')
     if os.path.exists(dest):
         logger.warning(f'Backup file already exists at {dest}, it will be overwritten.')
 
     try:
-        shutil.copy(source, dest)
+        shutil.copy2(source, dest)
```

Comment:
question: Just curious why `copy2` is needed, what's the difference between it and `copy`? 😝

Reply:
copy2 preserves the original file attributes and metadata. copy doesn't

Reply:
Aaah interesting, thanks! 🙂